### PR TITLE
add driftnet

### PR DESCRIPTION
**Pull request by Lajnux**
### Description:

added the hunter/fishing activity drift net fishing

### Changes:

added the hunter/fishing activity drift net fishing
Balanced rates and items needed

### Other checks:

-   [X] I have tested all my changes thoroughly.

### CODE REVIEW

File: src/commands/Minion/driftnet.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+			oneDriftNetTime = reduceNumByPercent(oneDriftNetTime, 30);
+			boosts.push('30% boost for using Stamina potion(4)');
+			itemsToRemove.add('Stamina potion(4)', 1);
+		}
+
+		const quantity = Math.round(tripLength / oneDriftNetTime);
+		const duration = quantity * oneDriftNetTime;
+
+		if (!bankHasItem(userBank.bank, itemID('Drift net'), quantity)) {
```
Comment (gc) [inline src/commands/Minion/driftnet.ts:110]: use `userBank.has()`

File: src/commands/Minion/driftnet.ts [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+		await msg.author.settings.sync(true);
+
+		if (msg.author.skillLevel(SkillsEnum.Fishing) < 47 || msg.author.skillLevel(SkillsEnum.Hunter) < 44) {
+			return msg.channel.send('You need atleast level 44 Hunter and 47 Fishing to do Drift net fishing.');
+		}
+
+		if (
+			!msg.author.hasItemEquippedAnywhere('Graceful gloves') ||
+			!msg.author.hasItemEquippedAnywhere('Graceful top') ||
+			!msg.author.hasItemEquippedAnywhere('Graceful legs')
```
Comment (gc) [inline src/commands/Minion/driftnet.ts:48]: You can check these in 1 line, as an array.

File: src/tasks/minions/HunterActivity/driftNetActivity.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+import { Bank } from 'oldschooljs';
+
+import { Activity } from '../../../lib/constants';
+import driftNetCreatures from '../../../lib/skilling/skills/hunter/driftNet';
+import { SkillsEnum } from '../../../lib/skilling/types';
+import { DriftNetActivityTaskOptions } from '../../../lib/types/minions';
+import { handleTripFinish } from '../../../lib/util/handleTripFinish';
+
+// Bonus loot from higher fishing level
+const FishBonusLoot = [
```
Comment (gc) [inline src/tasks/minions/HunterActivity/driftNetActivity.ts:12]: camelcase

**Review by gc — CHANGES_REQUESTED** [review]

File: src/commands/Minion/driftnet.ts [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+			tripTime = Math.floor(maxTripLength / Time.Minute);
+		}
+		await msg.author.settings.sync(true);
+
+		if (msg.author.skillLevel(SkillsEnum.Fishing) < 47 || msg.author.skillLevel(SkillsEnum.Hunter) < 44) {
+			return msg.channel.send('You need atleast level 44 Hunter and 47 Fishing to do Drift net fishing.');
+		}
+
+		if (!msg.author.hasItemEquippedAnywhere(['Graceful gloves', 'Graceful top', 'Graceful legs'])) {
+			return msg.channel.send('You need Graceful top,legs and gloves equipped to do Drift net fishing.');
```
Comment (gc) [inline src/commands/Minion/driftnet.ts:45]: missing a space here

File: src/commands/Minion/driftnet.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+		}
+
+		const quantity = Math.round(tripLength / oneDriftNetTime);
+		const duration = quantity * oneDriftNetTime;
+
+		itemsToRemove.add('Drift net', quantity);
+
+		if (!userBank.has(itemsToRemove.bank)) {
+			return msg.channel.send(
+				`You need ${quantity}x Drift net for the whole trip, try a lower trip lengthor make/buy more Drift net.`
```
Comment (gc) [inline src/commands/Minion/driftnet.ts:109]: missing a space here

File: src/tasks/minions/HunterActivity/driftNetActivity.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+			skillName: SkillsEnum.Hunter,
+			amount: huntXpReceived,
+			duration
+		})}`;
+		xpRes += `\n${await user.addXP({ skillName: SkillsEnum.Fishing, amount: fishXpReceived, duration })}`;
+		await user.incrementCreatureScore(fishShoal.id, fishShoalCaught);
+
+		let str = `${user}, ${user.minionName} finished drift net fishing and caught ${quantity}x ${fishShoal.name}. ${xpRes}\n${user.minionName} asks if you'd like them to do another of the same trip.`;
+
+		await user.addItemsToBank(loot.values(), true);
```
Comment (gc) [inline src/tasks/minions/HunterActivity/driftNetActivity.ts:78]: ii dont think you need loot.values() here, just loot.

**Review by gc — CHANGES_REQUESTED** [review]

**Review by themrrobert — APPROVED** [review]

**Pull request merged**